### PR TITLE
Refactor identity registry

### PR DESCRIPTION
**Pull request by ender503**
Add the TicIdentityRegistry class maintain the behavior of identity
registration and resolution, it is based on the Trusted IOTA Contacts,
and uses the master channel root to generate Decentralized Identifier,
and uses the profile channel to save the corresponding DID document.

See also: https://github.com/cr0ssing/tic.api.js

### CODE REVIEW

File: packages/did/src/mnid.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ var TRYTE = '9ABCDEFGHIJKLMNOPQRSTUVWXYZ';
 var tryte = require('base-x')(TRYTE);
 
 function checksum(payload) {
-  return new Buffer(sha3_256(Buffer.concat(payload)), 'hex').slice(0, 4);
+  return Buffer.from(sha3_256(Buffer.concat(payload)), 'hex').slice(0, 4);
```
Comment (jserv) [inline packages/did/src/mnid.js]: It is not related to the subject.

Comment (ender503) [reply packages/did/src/mnid.js]: Got it. 👌

**Comment by ender503** [comment]
Add the JSDoc to describe the `TicIdentityRegistry`.

File: packages/did/src/mnid.js (hunk 1 above)
Comment (jserv) [reply packages/did/src/mnid.js]: Move this change to #91

**Comment by jserv** [comment]
Why did you call it `TicIdentityRegistry` rather than plain `IdentityRegistry`?

**Comment by ender503** [comment]
I'm not sure which one would be better, so I renamed it to `TicIdentityRegistry`.
Look likes the `IdentityRegistry` is good enough.

File: packages/did/README.md [hunk 2]
```diff
@@ -25,19 +25,6 @@ const { register } = require('@tangleid/did');
...
-    'did:tangleid:MoWYKbBfezWbsTkYAngUu523F8YQgHfARhWWsTFSN2U45eAMpsSx3DnrV4SyZHCFuyDqjvQdg7',
-  masterRoot:
-    'OMEDQKSVKAFHEX9OSTBZLFBGXNMSUO9LEUFFCDTOSIJXCQMNFIHDCKHMUJCYPXGXXMPLHIAMLXBFAZADA',
-  password: 'THISISTHESEEDOFTHETICACCOUNTANDISHOULDNOTGIVEITTOANYBODYELSE',
-  profileRoot:
-    'YMGVIKHFZIGUVANVHIUAKJUTPAZKBBLATTCOPJJAQPIKNUNYCKKHHHSUULSBYJVXDXDGUJKMGSIVZSPPI',
-  contactsRoot:
-    'PUZSASSTMBDGPPYJJEZZODAIFWBMIMPMBLWNMOHWVUQHVSAHEKO9LHUSFUZYJZ9DNKLU9MF9FOYXEWLTK'
-}
-*/
```
Comment (ender503) [inline packages/did/README.md]: I updated the README.md to reflect the code changes.
This comment should not exist in here, so I remove it.

**Comment by jserv** [comment]
Rebasing is required.

**Comment by ender503** [comment]
Rebased.

**Pull request merged**